### ISSUE TITLE
Enhancement - doc2vec

### ISSUE THREAD

**Issue by waldren** [akb89/word2vec#3]
Hello,

I have been trying to convert a [simple doc2vec example](https://github.com/nfmcclure/tensorflow_cookbook/tree/master/07_Natural_Language_Processing/07_Sentiment_Analysis_With_Doc2Vec) to use tensorflow data pipeline. I was struggling to do the window sequence generation. I stumbled upon your project. I was hoping for some inspiration. Unfortunately my skill level is at least an order of magnitude less than yours in python/tf. So I was awed but not able to take away inspiration to solve my issue.

I wondered if there was any interest in adding a doc2vec train_mode?

**Comment by akb89** [comment]
Hi @waldren, 
Thanks for the suggestion. Unfortunately there is no plan to add support for doc2vec at the moment, although I do welcome pull requests! 
I feel you though, the window sequence generation part of the code has been really painful to write for me, and the code is particularly unreadable I have to admit. 
You probably already took a look but the package most interesting for you is probably [word2vec.utils.datasets](https://github.com/akb89/word2vec/blob/master/word2vec/utils/datasets.py) and especially the `get_w2v_train_dataset` function that details the whole w2v preprocessing pipeline, as well as the `extract_examples` function which is the one doing the window sequence generation for both cbow and skipgram in order to generate a Tensor of features and labels as expected by the tf.Dataset. Don't hesitate to look at the corresponding [unit tests](https://github.com/akb89/word2vec/blob/master/tests/utils_datasets_test.py) as well as those may provide useful examples to better understand the code. 
Hope that helps,

**Comment by waldren** [comment]
@akb89 I thought that would be the case, but thought I would ask :-) 

I have been able to dissect the code some and understand some of with working. I will fork the repo and if I can it working, I will definitely send a pull request!

Either way, your code has helped me better understand a working approach. I just saw the unit tests when I cloned the repo last night, so will definitely dig in there.

Greatly appreciate the response and the pointers.

Best,
Steven
